### PR TITLE
styling, markdown settings tied to config

### PR DESCRIPTION
**Pull request by jessopb**
links markdown settings to siteConfig
fixes some styling

### CODE REVIEW

File: client/src/components/FileViewer/index.jsx [hunk 1]
```diff
@@ -39,7 +40,7 @@ class FileViewer extends React.Component {
       <div className={'markdown'}>
         {
           this.state.fileLoaded &&
-          <ReactMarkdown className={'markdown-preview'} source={this.state.fileText} skipHtml />
+          <ReactMarkdown className={'markdown-preview'} source={this.state.fileText} skipHtml={skipHtmlMain} disallowedTypes={disallowedTypesMain} escapeHtml={escapeHtmlMain} />
```
Comment (skhameneh) [inline client/src/components/FileViewer/index.jsx]: This change is **very** dangerous, can you make a list of what is **whitelisted**?

Blacklists are not safe when it comes to script injection.

Comment (jessopb) [reply client/src/components/FileViewer/index.jsx]: It is now a very long whitelist :)

File: cli/defaults/siteConfig.json [hunk 2]
```diff
@@ -42,11 +42,43 @@
   },
   "serving": {
     "markdownSettings": {
-      "skipHtml": true,
-      "privilegedDisallowedTypesDescriptions": ["Image"],
-      "privilegedDisallowedTypesMain": [],
-      "publicDisallowedTypesDescriptions": ["Image"],
-      "publicDisallowedTypesMain": []
+      "skipHtmlMain": false,
+      "escapeHtmlMain": false,
```
Comment (skhameneh) [inline cli/defaults/siteConfig.json]: I took a closer look, set this to `true` and make sure it's documented as a dangerous setting.  
I don't want users deploying making mistakes, this needs to be explicitly `true` on all our instances that allow public content.

**Comment by jessopb** [comment]
I made the changes you requested. Also added an ErrorBoundary component to the FileViewer component to prevent it from crashing the app.

**Pull request merged**